webcompat/static/js/lib/bugform.js

```diff
@@ -573,6 +605,136 @@ function BugForm() {
     });
   };
 
+  // functions for Image Editor
+  this.openImageEditor = function() {
+    //var doc = global.document;
+    var canvas = document.createElement('canvas');
+    var ctx = canvas.getContext('2d');
+    var img = new Image();
+    canvas.id = "mCanvas";
+    canvas.width = 500;
+    canvas.height = 500;
+    img.src = this.dataURI;
+
+    var modalContent = document.querySelector('.modal-content');
+
+    this.editImageModal.removeClass("is-hidden");
+    modalContent.appendChild(canvas);
+
+    // TODO ask Mike if there is a better way to share this globally, like this.ctx
+    globalCtx = ctx;
+    globalCtx.drawImage(img, 0, 0);
+
+    // TODO ask Mike if there is a better way
+    // having to define the canvas event listeners here because at time
+    // of init definition, the canvas does not exist
+    var domCanvas = $("#mCanvas");
+    domCanvas.on("mousedown", _.bind(this.canvasMouseDown, this));
+    domCanvas.on("mousemove", _.bind(this.canvasMouseDrag, this));
+    domCanvas.on("mouseup", _.bind(this.canvasMouseUp, this));
+
+  }
+
+  this.closeImageEditor = function() {
+    var myCanvas = document.querySelector('canvas');
+ 
+    this.dataURI = myCanvas.toDataURL();
+    this.editImageModal.addClass("is-hidden");
+    this.addPreviewBackground(this.dataURI);
+  }
+
+  this.setImageEditorHighlight = function() {
+    this.curEditorActiveControl = this.ACTIVE_CONTROL_OPTIONS[0];
+  }
+
+  this.setImageEditorRedact = function() {
+    this.curEditorActiveControl = this.ACTIVE_CONTROL_OPTIONS[1];
+  }
+
+  this.canvasMouseDown = function(ev) {
+    var myCanvas = document.querySelector('canvas');
```

Comment:
You should be able to get the canvas element from the `ev` object, rather than needing to query the DOM again. Same for other places.